receiver/signalfxreceiver/receiver.go

```diff
@@ -165,17 +171,29 @@ func (r *sfxReceiver) Shutdown(context.Context) error {
 	return err
 }
 
-func (r *sfxReceiver) readBody(ctx context.Context, resp http.ResponseWriter, req *http.Request) ([]byte, bool) {
+func (r *sfxReceiver) readContentType(ctx context.Context, resp http.ResponseWriter, req *http.Request, events bool) (bool, bool) {
 	if req.Method != http.MethodPost {
 		r.failRequest(ctx, resp, http.StatusBadRequest, invalidMethodRespBody, nil)
-		return nil, false
+		return false, false
 	}
 
-	if req.Header.Get(httpContentTypeHeader) != protobufContentType {
+	otlpContent := false
+	switch req.Header.Get(httpContentTypeHeader) {
+	case protobufContentType:
+	case otlpProtobufContentType:
+		otlpContent = true
+		if events {
+			r.failRequest(ctx, resp, http.StatusUnsupportedMediaType, invalidEventsContentRespBody, nil)
+			return otlpContent, false
+		}
+	default:
 		r.failRequest(ctx, resp, http.StatusUnsupportedMediaType, invalidContentRespBody, nil)
-		return nil, false
+		return otlpContent, false
 	}
+	return otlpContent, true
+}
```

Comment:
Same here. No need to reuse this logic across `handleEventReq` and `handleDatapointReq`. Passing a boolean flag (like `events` in this case) that controls function's behavior is usually a code smell